MetaMask/metamask-mobile
# feat: Remove Account Snap Warning (Flask)

## **Description**

1. What is the reason for the change?
- This change adds an extra warning to users when trying to remove an account snap (https://metamask.github.io/snap-simple-keyring/latest/ for example) AND the user has snap accounts in their wallet
- Snap accounts are not managed/owned by metamask and therefore cannot be recovered in metamask. User's must manage and backup these accounts in the snaps UI.
2. What is the improvement/solution?
- Adds an extra layer of friction when trying to remove an account snap and the user has snap accounts in their wallet
- This is a `FLASK ONLY` feature

## **Related issues**

Fixes: https://github.com/MetaMask/accounts-planning/issues/155

## **Manual testing steps**

1. check this branch
2. open `.js.env`
3. ensure that `export METAMASK_BUILD_TYPE=flask` is set to flask (not main)
4. build the app on this branch
5. create/import a wallet
6. inside the mobile browser, open this URL: https://metamask.github.io/snap-simple-keyring/latest/
7. connect this snap
8. click create account, you can do this as many times as you want.
9. navigate back to the main wallet view
10. NOTICE there should be new accounts in your wallet with the name "Snap Account x"
11. Go to settings
12. navigate to snaps
13. locate the Snap simple keyring settings
14. scroll to the bottom of the page and click the red "Remove" button
15. Notice: new warning should pop up indicating that the user should backup these accounts before removing the snap
16. click continue
17. you should be prompted to type the name of the snap in the text input.
18. YOU SHOULD NOT BE ABLE TO REMOVE THE SNAP WITHOUT TYPING The CORRECT SNAP NAME
19. Once you type the correct snap name, you should be able to remove the snap
20. After clicking remove the snap should be removed.


#### Testing a non account snap
1. check this branch
2. open `.js.env`
3. ensure that `export METAMASK_BUILD_TYPE=flask` is set to flask (not main)
4. build the app on this branch
5. create/import a wallet
6. navigate to settings/snaps
7. click on the pre installed Message signing snap
8. click on the remove button at the bottom
9. `THERE SHOULD NOT BE A WARNING`

#### Testing removing a snap account without snap accounts in the wallet
1. check this branch
2. open `.js.env`
3. ensure that `export METAMASK_BUILD_TYPE=flask` is set to flask (not main)
4. build the app on this branch
5. create/import a wallet
6. inside the mobile browser, open this URL: https://metamask.github.io/snap-simple-keyring/latest/
7. connect this snap. Do not create any accounts this time.
8. Go to settings
9. navigate to snaps
10. locate the Snap simple keyring settings
11. scroll to the bottom of the page and click the red "Remove" button
12. There should be no warning pop up and the snap should be removed

## **Screenshots/Recordings**

### Extension Version

<img width="367" alt="Screenshot 2024-10-01 at 5 38 58 PM" src="https://github.com/user-attachments/assets/024e03f4-bf38-4062-a040-6eb89b1c39c9">
<img width="363" alt="Screenshot 2024-10-01 at 5 39 14 PM" src="https://github.com/user-attachments/assets/918dff2a-ca80-4bee-8889-28042606fe9f">
<img width="361" alt="Screenshot 2024-10-01 at 5 39 20 PM" src="https://github.com/user-attachments/assets/bf27f6bd-b859-49b8-ba84-3d35c399d77c">

### **After**


with snap accounts:

https://github.com/user-attachments/assets/24aa2993-98fa-4d72-bc1f-b7f9c1aeb6ad

<img width="825" alt="Screenshot 2024-10-10 at 3 26 46 PM" src="https://github.com/user-attachments/assets/0026de7c-e657-49ad-b6a6-e085ffc05816">


Without snap accounts:

https://github.com/user-attachments/assets/7cf42c03-dc6e-40fb-b490-fb0eb4d8ba83



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
